backend/internal/service/handler/reviews/update_review.go

```diff
@@ -0,0 +1,104 @@
+package reviews
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"platnm/internal/models"
+	"strconv"
+	"time"
+
+	"platnm/internal/constants"
+	"platnm/internal/errs"
+)
+
+type ReviewUpdate struct {
+	ID      int     `json:"id"`
+	UserID  string  `json:"user_id"`
+	Rating  *int    `json:"rating"`
+	Comment *string `json:"comment"`
+}
+
+func (h *Handler) UpdateReviewByReviewID(c *fiber.Ctx) error {
+
+	reviewIDStr := c.Params("id")
+
+	// Convert the string reviewID to an integer
+	reviewID, err := strconv.Atoi(reviewIDStr)
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid review ID"})
+	}
+
+	existingReview, err := h.reviewRepository.GetExistingReview(c.Context(), reviewIDStr)
+	reviewUpdate := ReviewUpdate{}
+
+	if err != nil {
+		return err
+	}
+
+	if existingReview == nil {
+		return errs.NotFound("Review", "id", reviewID)
+	}
+
+	// Validate the body of the request matches the form of ReviewUpdate
+	if err := c.BodyParser(&reviewUpdate); err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid request body"})
+	}
+
+	// Ensure at least one field is being updated
+	if reviewUpdate.Comment == nil && reviewUpdate.Rating == nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Either 'comment' or 'rating' must be provided"})
+	}
+
+	userID := reviewUpdate.UserID
+	reviewIsOwnedByUser, err := h.reviewRepository.ReviewBelongsToUser(c.Context(), reviewIDStr, userID)
+
+	if err != nil {
+		return err
+	}
+
+	if !reviewIsOwnedByUser {
+		// If no rows are returned, the review doesn't belong to the user
+		return errs.BadRequest("This user does not own the specified review.")
+	}
+
+	updateRating := reviewUpdate.Rating != nil
+	updateComment := reviewUpdate.Comment != nil
+
+	if !updateRating && updateComment {
+		existingReview = &models.Review{
```

Comment:
i believe this big if/else could be simplified since existingReview already has most of the fields populated from the db (all we really need to do here is overwrite rating and/or comment if provided). not a huge deal though, just wanted to flag it!